src/modules/search-time/selector/index.tsx

```diff
@@ -43,14 +43,52 @@ export default function SearchTimeSelector({
           };
 
     setSelectedMode(newState);
-
     onChange(newState);
   };
 
+  const addCurrentTimeToDate = (date: string) => {
+    return new Date(`${date} ${format(new Date(), 'HH:mm:ss')}`);
+  };
+
+  const resetToCurrentTime = () => {
+    setSelectedTime(() => format(new Date(), 'HH:mm'));
+  };
+
+  const resetToCurrentDate = () => {
+    setSelectedDate(new Date());
+  };
+
+  const addCurrentDateToTime = (time: string) => {
+    const [hours, minutes] = time.split(':').map(Number);
+    const now = new Date();
+    now.setHours(hours, minutes, 0, 0);
+    return now;
+  };
+
+  const isPastDate = (selectedDate: string) => {
+    const today = new Date().toISOString().split('T')[0];
+
+    // If reselecting from (selectedYear > current year) to current year, the date will autmatically be reset to the current date.
+    if (selectedDate.substring(0, 4) < initialDate.getFullYear().toString()) {
+      resetToCurrentDate();
+    }
+
+    // To ensure that the time is not past in whenever reselecting back the current date.
+    if (selectedDate <= today) resetToCurrentTime();
+    return selectedDate < today;
+  };
+
+  const isPastTime = (time: string) => {
+    if (new Date() < selectedDate) return false;
+    const now = addCurrentDateToTime(time);
+    return now < selectedDate;
+  };
+
   const internalOnDateChange = (event: ChangeEvent<HTMLInputElement>) => {
     if (!event.target.value) return;
+    if (isPastDate(event.target.value)) return;
 
-    setSelectedDate(new Date(event.target.value));
+    setSelectedDate(addCurrentTimeToDate(event.target.value));
```

Comment:
Should this be `addCurrentDateToTime`?

Reply:
It seems that whether new Date() or addCurrentTimeToDate() is used here actually does not affect the behavior of the component. I can change back to using new Date(), as addCurrentTimeToDate() is excessive and adds unecessery complexity.

Reply:
Agree. The date input only cares about the first date part of the Date, so the time does not matter here.